addons/binding/org.openhab.binding.wso2iots/src/main/java/org/openhab/binding/wso2iots/handler/BridgeHandler.java

```diff
@@ -0,0 +1,219 @@
+
+        logger.debug("Initializing bridge handler.");
+
+        BridgeConfiguration config = getConfigAs(BridgeConfiguration.class);
+        logger.debug("config hostname = {}", config.hostname);
+        logger.debug("config port = {}", config.port);
+
+        boolean validConfig = true;
+        String errorMsg = null;
+
+        if (StringUtils.trimToNull(config.hostname) == null) {
+            errorMsg = "Parameter 'hostname' is mandatory and must be configured";
+            validConfig = false;
+        }
+        if (config.port == null) {
+            errorMsg = "Parameter 'port' must be configured";
+            validConfig = false;
+        }
+        if (config.refresh == null) {
+            errorMsg = "Parameter 'refresh' must be configured";
+            validConfig = false;
+        }
+
+        if (validConfig) {
+            updateStatus(ThingStatus.OFFLINE);
+            startAutomaticRefresh(config);
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, errorMsg);
+        }
+    }
+
+    @Override
+    public void dispose() {
+        logger.debug("Disposing the bridge handler.");
+
+        if (refreshJob != null && !refreshJob.isCancelled()) {
+            refreshJob.cancel(true);
+            refreshJob = null;
+        }
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // No channels - nothing to do
+    }
+
+    /**
+     * Start the job refreshing the handler
+     */
+    private void startAutomaticRefresh(BridgeConfiguration config) {
+        if (refreshJob == null || refreshJob.isCancelled()) {
+            Runnable runnable = () -> {
+                try {
+
+                    checkConnection();
+
+                } catch (Exception e) {
+                    logger.error("Exception occurred during execution: {}", e.getMessage(), e);
+                }
+            };
+
+            int delay = config.refresh.intValue();
+            refreshJob = scheduler.scheduleWithFixedDelay(runnable, 0, delay, TimeUnit.MINUTES);
+        }
+    }
+
+    protected BridgeConfiguration getBridgeConfiguration() {
+
+        BridgeConfiguration config = getConfigAs(BridgeConfiguration.class);
+        return config;
+
+    }
+
+    private void checkConnection() throws Exception {
+
+        String errormsg = null;
+        String response1 = null;
+        BridgeConfiguration config = getConfigAs(BridgeConfiguration.class);
+        String hostname = config.hostname;
+
+        String url = "http://" + hostname + ":9763/services/Version.VersionHttpEndpoint";
+
+        URL obj;
+
+        obj = this.validateCertificate(url);
+
+        if (obj == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
+                    "Certificate validation failed");
+
+        } else {
+
+            HttpURLConnection con = (HttpURLConnection) obj.openConnection();
```

Comment:
We have some proven technology for this in eclipse smarthome, please use that instead. 

https://www.eclipse.org/smarthome/documentation/javadoc/org/eclipse/smarthome/io/net/http/HttpUtil.html

Note that a lot of potential corner cases with closing connections and handling exceptions exist in such communication and are handled properly by the util.

Note that it also contains a builder like structure which is less cryptic, see an example in https://github.com/openhab/openhab2-addons/blob/116ba76b3de9b1d5e166d64b85f7272cc7bfb3f2/addons/binding/org.openhab.binding.icloud/src/main/java/org/openhab/binding/icloud/internal/ICloudConnection.java